ampd/src/solana/msg_verifier.rs

```diff
@@ -1,23 +1,23 @@
 use axelar_wasm_std::voting::Vote;
-use gmp_gateway::events::GatewayEvent;
+use gmp_gateway::events::{CallContract, GatewayEvent};
 use solana_transaction_status::{
     option_serializer::OptionSerializer, EncodedConfirmedTransactionWithStatusMeta,
 };
-use std::sync::Arc;
+use std::{borrow::Cow, sync::Arc};
 use tracing::error;
 
 use crate::handlers::solana_verify_msg::Message;
 
-impl PartialEq<&Message> for GatewayEvent {
+impl PartialEq<&Message> for GatewayEvent<'_> {
     fn eq(&self, msg: &&Message) -> bool {
         match self {
-            GatewayEvent::CallContract {
+            GatewayEvent::CallContract(Cow::Owned(CallContract {
```

Comment:
There's another variant on this enum that is `GatewayEvent::MessageApproved` . Should we use that instead of `GatewayEvent::CallContract` ? I appreciate an explanation 🙏